### PR TITLE
feat: implement semantic versioning modes and conventional commits support

### LINKED ISSUE #264: not fully conventional commit compliant
if you commit `feat!: foo bar` is not trying to bump a major unless
you define explicitly `major-keywords: feat!`

it is fine to set it but I would expect the GH action is fully compliant with https://www.conventionalcommits.org/en/v1.0.0/

### PR DESCRIPTION
**Pull request by virgofx**
### Summary

Implements issue #264 by adding a configurable `semver-mode` input that introduces Conventional Commits parsing as the new default, while preserving the legacy keyword-based detection as an explicit opt-in mode. The initially proposed `conventional-commits-preset` input (`conventionalcommits` | `angular`) was added then removed after analysis showed both presets produce identical results for our use case — simplifying the interface to a single `semver-mode` toggle.

### What Changed

**New action input**

- `semver-mode` (`conventional-commits` | `keywords`) — controls how semantic version bumps are determined from commit messages. Defaults to `conventional-commits`.

**New module: `src/commit-analyzer.ts`**

- Unified release-type analysis module with `parseConventionalCommit()` and `detectConventionalCommitReleaseType()`.
- Parser options taken verbatim from the [`conventional-changelog-conventionalcommits` preset](https://github.com/conventional-changelog/conventional-changelog/blob/master/packages/conventional-changelog-conventionalcommits/src/parser.js) (`headerPattern`, `breakingHeaderPattern`, `headerCorrespondence`, `noteKeywords`, `revertPattern`, `issuePrefixes`).
- Supports the full Conventional Commits v1.0.0 spec: `!` breaking indicator in header, `BREAKING CHANGE` / `BREAKING-CHANGE` footer tokens, scoped types.
- New `conventional-commits-parser` dependency (the only new runtime dep).

**Refactored release detection**

- `src/terraform-module.ts` — `computeReleaseType()` now delegates to `commit-analyzer.ts` based on `config.semverMode`, replacing inline keyword logic for the conventional-commits path.
- Removed unused `releaseType` imports from `src/releases.ts`, `src/tags.ts`, `src/pull-request.ts`.

**Configuration & types**

- Added `SemverMode` type, `VALID_SEMVER_MODES` constant, `semverMode` to `Config` interface.
- Added `ConventionalCommitResult` type in new `src/types/commit-analyzer.types.ts`.
- Updated `src/config.ts` with `semverMode` validation.
- Updated `src/utils/metadata.ts` with new input mapping.
- Removed the `conventional-commits-preset` input that was added then dropped (preset concept fully removed — angular and conventionalcommits presets were functionally identical).

**Documentation**

- `README.md` — Added `semver-mode` to inputs table, new "Type-to-Release Mapping" section, "Breaking Change Notation" section explaining `!` vs footer syntax.
- `action.yml` — Added `semver-mode` input with detailed description; updated keyword input descriptions to note they only apply in `keywords` mode.
- `docs/testing.md`, agent instructions, `AGENTS.md` — Updated to reflect new module and metadata sync requirements.

**SonarQube**

- Added `sonar.issue.ignore.multicriteria` exclusion for `typescript:S5852` (regex backtracking) scoped to `src/commit-analyzer.ts` — the patterns are upstream verbatim and only process short commit headers.

**Tests**

- New `__tests__/commit-analyzer.test.ts` — 108 tests covering single-message parsing, release-type detection (major/minor/patch/null), edge cases (empty descriptions, multiline bodies, special characters), and full `computeReleaseType()` integration.
- Updated `__tests__/config.test.ts` — validation tests for `semverMode`.
- Updated `__tests__/terraform-module.test.ts` — refactored to test both `conventional-commits` and `keywords` modes.
- Updated `__tests__/utils/metadata.test.ts` — stricter mapping assertions to prevent input drift.

### Validation

- `npm run check:fix` — clean
- `npm run textlint:fix` — clean
- `npm run typecheck` — clean
- `npm run test` — **622/622 tests passing, 100% coverage** (statements/branches/functions/lines)

### Risks / Notes

- **BREAKING CHANGE**: Default semver detection now uses Conventional Commits parsing (`semver-mode: conventional-commits`) instead of keyword matching. Legacy behavior remains available via `semver-mode: keywords`.
- In practice, most consumers using defaults should not be impacted unless they rely on non-conventional commit message patterns — in which case they can set `semver-mode: keywords` to restore prior behavior.
- The `conventional-commits-preset` input was intentionally not shipped — analysis showed angular and conventionalcommits presets produce identical release-type results, making the toggle unnecessary complexity.

Fixes #264

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -184,26 +184,105 @@ resources.
...
+| `terraform-docs-version`         | Specifies the terraform-docs version used to generate documentation for the wiki                                                                                                                                                                                                                                                                                                                                                                                                                                                                            | `v0.19.0`                                                                                             |
```
[the remark this answers is not in the document]
Comment (virgofx) [reply README.md]: @copilot bump to v0.21.0 (latest)
